linq4j/src/test/java/org/apache/calcite/linq4j/test/DeterministicTest.java

```diff
@@ -423,15 +467,15 @@ private boolean isConstant(Expression e) {
         equalTo("{\n"
             + "  return new Runnable(){\n"
             + "      int test() {\n"
-            + "        return "
-            + "$L4J$C$java_math_BigInteger_valueOf_42L_add_java_math_BigInteger_valued8d57d69;\n"
+            + "        return ((java.math.BigInteger) (java.math.BigInteger) $L4J$C$java_math_BigInteger_valueOf_42L_.get()).add((java.math.BigInteger) (java.math.BigInteger) $L4J$C$java_math_BigInteger_valueOf_42L_.get());\n"
```

Comment:
Note: previously `BigInteger.valueOf(42).add(BigInteger.valueOf(42))` was memoized, however now you perform `.add` in the runtime.

Could you please keep that optimization in place?

Reply:
So sorry for this mistake, fixed now.